### PR TITLE
[Luau] Add new port to luau

### PR DESCRIPTION
**Pull request by sssooonnnggg**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/luau/cmake-config-export.patch [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
++  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
++)
++
++if (LUAU_BUILD_CLI)
++    install(
++        TARGETS Luau.Repl.CLI
++        EXPORT luau-targets
++        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
++    )
++endif()
```
Comment (FrankXie05) [inline ports/luau/cmake-config-export.patch:109]: If it is exported by vcpkg we should add the `unofficial` prefix. However, I think this patch can ask upstream for approval. :)

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

File: ports/luau/cmake-config-export.patch (hunk 1 above)
Comment (dg0yt) [reply ports/luau/cmake-config-export.patch:109]: > this patch can ask upstream for approval. :)

No. There are 3 pending PRs to add install commands, including one which adds exported config.

Comment (sssooonnnggg) [reply ports/luau/cmake-config-export.patch]: The official response time seems a bit slow, I will first add the unofficial namespace and then try to submit an official PR.

**Comment by sssooonnnggg** [comment]
Thanks for the review，make namespace unofficial now. If the official repo adds official exports in the future, we can remove this patch. cc @FrankXie05 @dg0yt

**Comment by dg0yt** [comment]
Required package name: `unofficial-luau`
Required namespace: `unofficial::luau::`

FTR I wonder if this can be installed simultaneously with lua or luajit.

**Comment by sssooonnnggg** [comment]
@dg0yt Yes, I have tested luau can be installed simultaneously with lua and luajit. There libs has different names and header are splitted into different sub folders:
- lua => include/...
- luau => include/luau
- luajit => includde/luajit/...

**Comment by sssooonnnggg** [comment]
Hi, can you help with reviewing again? 😊 @FrankXie05

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
The usage has been tested successfully locally on `x64-windows`.

**Pull request merged**